noir-projects/aztec-nr/compressed-string/src/field_compressed_string.nr

```diff
@@ -1,11 +1,24 @@
-use dep::aztec::protocol_types::{traits::{Deserialize, Serialize}, utils::field::field_from_bytes};
+use dep::aztec::protocol_types::{
+    traits::{Deserialize, Packable, Serialize},
+    utils::field::field_from_bytes,
+};
 
 // A Fixedsize Compressed String.
 // Essentially a special version of Compressed String for practical use.
 pub struct FieldCompressedString {
     value: Field,
 }
 
+impl Packable<1> for FieldCompressedString {
+    fn pack(self) -> [Field; 1] {
+        [self.value]
+    }
+
+    fn unpack(input: [Field; 1]) -> Self {
+        Self { value: input[0] }
+    }
+}
+
```

Comment:
Why not use serialize here

Reply:
Because we now need to implement `Packable` for types that are to be stored in public storage.

Should have put there a comment. Did that in [this commit](https://github.com/AztecProtocol/aztec-packages/pull/11136/commits/a0e1427da41d4c88971f002d97b772ecbf748940) + used there the `#[derive(Serialize, Deserialize)]` macro and made the Packable impl. use it.

Reply:
Yes sorry, I meant doing exactly what you did - derive Ser and and use for packable (which is what deriving packable will do)